packages/keyvAdapter/src/index.ts

```diff
@@ -0,0 +1,35 @@
+import type {
+  KeyValueCache,
+  KeyValueCacheSetOptions,
+} from "@apollo/utils.keyvaluecache";
+import Keyv from "keyv";
+
+export class KeyvAdapter<V = string> implements KeyValueCache<V> {
+  private readonly keyv: Keyv<V>;
+
+  constructor(keyv?: Keyv<V>) {
+    this.keyv = keyv ?? new Keyv<V>();
+  }
+
+  async get(key: string): Promise<V | undefined> {
+    return this.keyv.get(key);
+  }
+
+  async set(
+    key: string,
+    value: V,
+    opts?: KeyValueCacheSetOptions,
+  ): Promise<void> {
+    // Maybe an unnecessary precaution, just being careful with 0 here. Keyv
+    // currently handles 0 as `undefined`. Also `NaN` is typeof `number`
+    if (typeof opts?.ttl === "number" && !Number.isNaN(opts.ttl)) {
+      this.keyv.set(key, value, opts.ttl * 1000);
```

Comment:
These are missing awaits (or returns), no?

Reply:
Yep, good catch. I tried to write a test that failed due to the missing awaits (like trying to get something immediately after setting, but I have to await the `get` which probably allows the unawaited `set` to resolve in time). Any ideas on a test that would fail for this mistake (that's resolved by awaiting)?

Reply:
You could implement your own keyv Store whose `set` awaits setImmediate or a short timeout before doing something observable?

Reply:
🎯 [9573b57](https://github.com/apollographql/apollo-utils/pull/93/commits/9573b57f6dcd412981385241b6ed2f3dc7d4ef7b)

(refixed in follow up commit [3f1d270](https://github.com/apollographql/apollo-utils/pull/93/commits/3f1d2702149b5fec7fa6e08796322a1d9e128208), tests passing)